ui/__tests__/exercise.test.js

```diff
@@ -0,0 +1,163 @@
+      wrapper.vm.sortBy("name");
+      wrapper.vm.sortBy("name");
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises[0].name).toBe("Leg Day");
+    });
+
+    test("should sort by primary_body_part in ascending order", async () => {
+      wrapper.vm.sortBy("primary_body_part");
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises[0].primary_body_part).toBe("Biceps");
+    });
+
+    test("should sort by primary_body_part in descending order", async () => {
+      wrapper.vm.sortBy("primary_body_part");
+      wrapper.vm.sortBy("primary_body_part");
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises[0].primary_body_part).toBe("Triceps");
+    });
+
+    test("should sort by secondary_body_part in ascending order", async () => {
+      wrapper.vm.sortBy("secondary_body_part");
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises[0].secondary_body_part).toBe(
+        "Calves"
+      );
+    });
+
+    test("should sort by secondary_body_part in descending order", async () => {
+      wrapper.vm.sortBy("secondary_body_part");
+      wrapper.vm.sortBy("secondary_body_part");
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises[0].secondary_body_part).toBe(
+        "Triceps"
+      );
+    });
+  });
+  describe("Search functionality", () => {
+    test("should return an empty list if no match is found", async () => {
+      wrapper.setData({ searchQuery: "Nonexistent Exercise" });
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises.length).toBe(0);
+    });
+
+    test("should filter workouts case-insensitively", async () => {
+      wrapper.setData({ searchQuery: "lEg DaY" });
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises.length).toBe(1);
+      expect(wrapper.vm.filteredExercises[0].name).toBe("Leg Day");
+    });
+    test("should filter exercises by name", async () => {
+      wrapper.setData({ searchQuery: "Leg Day" });
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises.length).toBe(1);
+      expect(wrapper.vm.filteredExercises[0].name).toBe("Leg Day");
+    });
+
+    test("should filter exercises by primary_body_part", async () => {
+      wrapper.setData({ searchQuery: "Biceps" });
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises.length).toBe(1);
+      expect(wrapper.vm.filteredExercises[0].primary_body_part).toBe("Biceps");
+    });
+
+    test("should filter exercises by secondary_body_part", async () => {
+      wrapper.setData({ searchQuery: "Triceps" });
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.filteredExercises.length).toBe(2);
+      expect(wrapper.vm.filteredExercises[0].secondary_body_part).toBe(
+        "Calves"
+      );
+      expect(wrapper.vm.filteredExercises[1].secondary_body_part).toBe(
+        "Triceps"
+      );
+    });
+  });
```

Comment:
this should filter by all fields - update test name

Reply:
done __ "should filter exercises across all fields"